src/ipc/ipc4/dai.c

```diff
@@ -367,6 +318,10 @@ int dai_config(struct comp_dev *dev, struct ipc_config_dai *common_config,
 		}
 	}
 
+	ret = dai_init_llp_info(dev);
```

Comment:
Is it safe to find the free one here and mark it as used later inside a call to dai_dma_position_init()? No double booking by another context that may pre-empt this one in between guaranteed?

Reply:
thanks, add spin lock to protect it

Reply:
if this is only called from IPC context, then this is already implicitly protected by IPC serialisation

Reply:
@lyakh   A genius idea!  llp slot is only accessed from IPC context like initializing module or stopping pipepline by host, so it is by IPC serialization in host. Thanks @mmaka1 and lyakh, it is a good lesson.